src/axom/core/numerics/matvecops.hpp

```diff
@@ -571,9 +575,11 @@ inline bool linspace( const T& x0, const T& x1, T* v, int N )
 template < typename T >
 inline void cross_product( const T* u, const T* v, T* w )
 {
-  assert( "pre: u pointer is null" && (u != nullptr) );
-  assert( "pre: v pointer is null" && (v != nullptr) );
-  assert( "pre: w pointer is null" && (w != nullptr) );
+  #if !defined(__CUDA_ARCH__)
```

Comment:
Instead of using guards with `__CUDA_ARCH__`, I suggest we add the following macro in `axom/Macros.hpp`:
```
#if defined(__CUDA__ARCH__)
#define AXOM_DEVICE_CODE
#endif 
```

That way this can be extended to different device types in one place while we more uniformly use `AXOM_DEVICE_CODE` throughout the code.

Reply:
Added.